### ISSUE TITLE
Lightbox AJAX content with link

### ISSUE THREAD

**Issue by gdanova** [wet-boew/wet-boew#7639]
Anyone know if there is a way to use Lightbox to open content as follows:

I’m currently using [AJAX - Content filtered by URL hash](http://wet-boew.github.io/v4.0-ci/demos/lightbox/lightbox-en.html#examples1) but in the file that loads in Lilghtbox, there’s **another Lightbox link** to an anchor in that same file, butthe link doesn’t work.

For example, page 1 has:

`<a class="wb-lbx" href="page2.html#link1">link 1</a>`

The content on page 2 that opens in Lightbox is:

```
<section id="link1" class="modal-dialog modal-content overlay-def">
   <header class="modal-header">
       <h2 class="modal-title">Heading</h2>
   </header>
   <div class="modal-body">
      <p>Content of link 1 with <a class="wb-lbx" href="#link2">link 2</a> to another piece of content on this same page.</p>
   </div>
</section>
```

This second link doesn’t work.

It’s not just an issue of a Lightbox link containing another Lightbox link that not working, because if everything is on a single page with `mfp-hide` then you _can_ continuously link from Lightbox content to Lightbox content. But because I want to link to page 2, it doesn’t work.

Any insight would be appreciated.

**Comment by sgdowney** [comment]
Most likely the second Lightbox link is not being enhanced/activated because it is being loaded after the wb enhancements have been applied to the page.  You would need to manually initialize any Lightbox links that exist in the file being ajaxed in.

**Comment by gdanova** [comment]
From the documentation, I assume that has to do with `wb-init.wb-lbx`, as in 

> Triggered manually (e.g., $( ".wb-lbx" ).trigger( "wb-init.wb-lbx" );).

 but I don't know how to add that to my 2nd page.

**Comment by hsrudnicki** [comment]
Does this remain still an issue? Please update the status of this again, whether this can be closed or otherwise, thank you

**Comment by kpduncan** [comment]
I was able to load two modals from a different page using this method:

Index Page:

`<a id="button1" class="btn btn-warning wb-lbx" href="modals.html#modal1">Modal 1</a>`
`<a id="button2" class="btn btn-warning wb-lbx" href="modals.html#modal2">Modal 2</a>`

Modals Page:

`<section id="modal1" class="modal-dialog modal-content overlay-def">
	<header class="modal-header">
		<h2 class="modal-title">Modal 1</h2>
	</header>
	<div class="modal-body">
		<button class="btn btn-warning modal2">Modal 2</button>
	</div>
</section>`

`<section id="modal2" class="modal-dialog modal-content overlay-def">
	<header class="modal-header">
		<h2 class="modal-title">Modal 2</h2>
	</header>
	<div class="modal-body">
		<p>..</p>
	</div>
</section>`

Javascript (add to end of index page):

`<script>
$(document).on("click", ".link2", function(){
	$('#button2').trigger('click');
});
</script>`

This seems to work for me.

**Comment by gdanova** [comment]
That's not quite what I'm trying to achieve. Linking to two modals independently works fine, but if one of the modals contains a link to another modal, it doesn't work.

```
<section id="link1" class="modal-dialog modal-content overlay-def">
   <header class="modal-header">
       <h2 class="modal-title">Heading</h2>
   </header>
   <div class="modal-body">
      <p>Content of link 1 with <a class="wb-lbx" href="#link2">link 2</a> to another piece of content on this same page.</p>
   </div>
</section>
```

**Comment by kpduncan** [comment]
Yes, I understand that, but during my testing the issue is when you click this

`<a class="wb-lbx" href="page2.html#link1">link 1</a>`

It not taking you too that page instead it generating the modal on the current page. The trigger method is the page as it is looking for the modal on the current page. The code I provide essentially gives you a loop whole around this issue.

**Comment by hsrudnicki** [comment]
Can be this closed as resolved or otherwise? Please update the status of this again, thx

**Comment by gdanova** [comment]
I understand the essence, not the details, of your workaround. Unfortunately, I can't incorporate that.

It's strange because whether I use:

`<p>Content of link 1 with <a class="wb-lbx" href="#link2">link 2</a> to another piece of content on this same page.</p>`
or
`<p>Content of link 1 with <a class="wb-lbx" href="page2.html#link2">link 2</a> to another piece of content on this same page.</p>`

...the result is the same. Links works, but no Lightbox. Still looking for a usable solution.

Thx.

**Comment by kpduncan** [comment]
Hey sorry if my exaptation didn't provide enough details. I recently just started providing help on WET issues as I have my own blog that uses it and believe in the WET bootstrap project. As a recent graduate as a computer programmer this is an excellent way to fine tune my skills and help others. Please review this live demo on my blog [here](https://www.cybersubtype.ca/apps/wet-demos/).

**Comment by gdanova** [comment]
By "I understand the essence, not the details, of your workaround", I did not mean that you didn't provide enough details. I just meant I don't understand the details because I'm not a programmer. :-)

But I'll check out those references anyway,
